clang-tools-extra/clang-tidy/hooks/GuardCallNonConstCheck.cpp

```diff
@@ -0,0 +1,74 @@
+//===--- GuardCallNonConstCheck.cpp - clang-tidy --------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "GuardCallNonConstCheck.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang {
+namespace tidy {
+namespace hooks {
+
+namespace {
+
+auto guardCall() {
+  return callExpr(callee(functionDecl(
+      hasName("_g"))),
+      hasArgument(0, expr().bind("guardIdExpr")),
+      hasArgument(1, expr().bind("guardLimitExpr"))
+  );
+}
+
+}
+
+void GuardCallNonConstCheck::registerMatchers(MatchFinder *Finder) {
+  Finder->addMatcher(expr(guardCall()).bind("unsupportedGuardedCall"), this);
```

Comment:
not "unsupported" again (I think I've commented on this line before, but now I don't se it)...